roles/st2/tasks/main.yml

```diff
@@ -1,22 +1,6 @@
 ---
 
-- name: Install latest st2 package
-  become: yes
-  apt:
-    name: st2
-    state: latest
-  when: st2_version == "stable"
-  notify:
-    - restart st2
-
-- name: Install latest st2 package
-  become: yes
-  apt:
-    name: st2={{ st2_version }}-{{ st2_revision }}
-    state: present
-  when: st2_version != "stable"
-  notify:
-    - restart st2
+- include: "{{ ansible_os_family }}.yml"
```

Comment:
Since it only about installing package, you can avoid splitting into `{{ ansible_os_family }}.yml` here, just use `package` module. http://docs.ansible.com/ansible/package_module.html

See solution in `st2mistral` by @humblearner https://github.com/StackStorm/ansible-st2/blob/1426c6d28184dc61f6d41cb7b2731966d331d3bf/roles/st2mistral/tasks/main.yml#L10-L24 which is pretty elegant approach.